### PR TITLE
Rename Blocks Purger to Tenant Deletion.

### PR DESCRIPTION
**Pull request by pstibrany**
**What this PR does**: This PR just renames "Blocks Purger" to "Tenant Deletion". It doesn't do anything else. Really. I hope.

**Checklist**
- [x] Tests updated
- [na] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/cortex/modules.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ const (
 	StoreGateway             string = "store-gateway"
 	MemberlistKV             string = "memberlist-kv"
 	ChunksPurger             string = "chunks-purger"
-	BlocksPurger             string = "blocks-purger"
+	TenantDeletion           string = "blocks-purger"
```
Comment (pracucci) [inline pkg/cortex/modules.go:78]: Can we also safely rename the value `blocks-purger`?

**Review by pracucci — APPROVED** [review]
LGTM with a question

File: pkg/cortex/modules.go (hunk 1 above)
Comment (pstibrany) [reply pkg/cortex/modules.go]: Yes. Given that it's marked as "invisible" to user and thus considered internal, I think that's perfectly fine.

**Comment by bboreham** [comment]
#4370 seems to be undoing this rename.  Can you explain why it was done please, to help judge whether undoing it is a good idea?

**Comment by pstibrany** [comment]
This PR was done because "blocks-purger" wasn't doing any "block purging". All "blocks purger" was doing was to accept API request to delete a tenant, and store marker file to the bucket. Hence "TenantDeletionAPI" seemed like better description.

Blocks cleanup logic itself is implemented in compactor instead.

**Pull request merged**